components/Base/Button.vue

```diff
@@ -6,55 +6,42 @@ import ArrowRight from '~/assets/icons/arrow-right.svg?skipsvgo';
 
 const props = withDefaults(
   defineProps<{
-    cta?: boolean;
-    variant?: 'fill' | 'outline' | 'featured-link';
-    type?: 'regular' | 'round' | 'square';
+    variant?:
+      | 'fill'
+      | 'outline'
+      | 'text'
+      | 'featured-link'
+      | 'icon-square'
+      | 'icon-round';
+    size?: 'medium' | 'big' | 'round' | 'square';
```

Comment:
I think we don't need `round` and `square` sizes as they should be covered by variants 👌

Reply:
Yes but then I have to have cross over. I would have to variant for round + fill and variant for round + outline combination which i don't think is a good solution. This way have a clear separation of concern. One prop controlling size/shape and controlling variant aka look and feel, wdyt?

Reply:
I think this doesn't really make sense to use something like `size="round"`. Maybe we just need a separate property for `round` and `square` values 🤔